packages/market/src/blockchain-facade/Supply.ts

```diff
@@ -92,12 +92,21 @@ export const createSupply = async (
 
     let { url, propertiesDocumentHash } = supplyPropertiesOnChain;
 
-    if (configuration.offChainDataSource) {
-        url = supply.getUrl();
-        propertiesDocumentHash = offChainStorageProperties.rootHash;
-    }
+    url = supply.getUrl();
+    propertiesDocumentHash = offChainStorageProperties.rootHash;
+
+    let idExists = false;
+    do {
```

Comment:
Such a construct has a potential to put the system on freeze, infinite loop when for some reason offchain service is down. 

The recommended way is to use tools like poly-js with exponential time ramp.